core/inflation/src/lib.rs

```diff
@@ -0,0 +1,90 @@
+// This file is part of Darwinia.
+//
+// Copyright (C) 2018-2022 Darwinia Network
+// SPDX-License-Identifier: GPL-3.0
+//
+// Darwinia is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Darwinia is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Darwinia. If not, see <https://www.gnu.org/licenses/>.
+
+//! Darwinia economic inflation mechanism implementation.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![deny(missing_docs)]
+
+#[cfg(test)]
+mod test;
+
+// crates.io
+use primitive_types::U256;
+// darwinia
+use dc_types::{Balance, Moment, UNIT};
+// github
+use substrate_fixed::{
+	transcendental,
+	types::{I95F33, U94F34},
+};
+
+/// Inflation's upper limit.
+pub const TOTAL_SUPPLY: Balance = 10_000_000_000 * UNIT;
+
+/// Milliseconds per year for the Julian year (365.25 days).
+pub const MILLISECS_PER_YEAR: Balance = (366 * 24 * 60 * 60) * 1000;
```

Comment:
Different with comment? But it seems to be inherited from current code.

Reply:
Yep. Just a copy of the old code.